one/src/ethereum_rpc.rs

```diff
@@ -0,0 +1,175 @@
+                "jsonrpc": "2.0",
+                "method": "eth_getTransactionByHash",
+                "params": [transaction_hash],
+                "id": 1,
+            }))
+            .send()
+            .await?
+            .json()
+            .await?)
+    }
+}
+
+impl EthRpc for HttpEthRpc {
+    async fn root_time_by_transaction_cid(&self, cid: Cid) -> Result<RootTime> {
+        // transaction to blockHash, blockNumber, input
+        let tx_hash = format!("0x{}", hex::encode(cid.hash().digest()));
+        let json: serde_json::Value = self.eth_transaction_by_hash(&tx_hash).await?;
+        debug!("txByHash response: {}", json);
+        let result = json.get("result").context("missing field result")?;
+        let block_hash = result
+            .get("blockHash")
+            .context("missing field result.blockHash")?;
+        let block_hash = block_hash
+            .as_str()
+            .context("missing field result.blockHash was not a string")?;
+        let block_number = result
+            .get("blockNumber")
+            .context("missing field result.blockNumber")?;
+        let block_number = block_number
+            .as_str()
+            .context("missing field result.blockNumber was not a string")?;
+        let block_number = get_timestamp_from_hex_string(block_number)
+            .context("missing field result.blockNumber was not a hex timestamp")?;
+        let input = result.get("input").context("missing field result.input")?;
+        let input = input
+            .as_str()
+            .context("missing field result.input was not a string")?;
+        let root = get_root_from_input(input)?;
+
+        // Get the block with block_hash if latest block number minus result.number grater then 3 the block is stable.
+        let Some(Ok(latest_block_number)) = self.eth_block_number().await?["result"]
+            .as_str()
+            .map(get_timestamp_from_hex_string)
+        else {
+            return Err(anyhow!("latest_block_number not found"));
+        };
+        if latest_block_number - block_number < 3 {
+            return Err(anyhow!("latest_block_number - block_number < 3"));
+        }
+
+        // Get the block time.
+        let json: serde_json::Value = self.eth_block_by_hash(block_hash).await?;
+        debug!("blockByHash response: {}", json);
+
+        let Some(timestamp) = json["result"]["timestamp"].as_str() else {
+            return Err(anyhow!(
+                "missing field result.timestamp or was not a string"
+            ));
+        };
+        let timestamp = get_timestamp_from_hex_string(timestamp)?;
+        Ok(RootTime {
+            root,
+            block_hash: block_hash.to_string(),
+            timestamp,
+        })
+    }
+}
+
+fn get_root_from_input(input: &str) -> Result<Cid> {
+    if let Some(input) = input.strip_prefix("0x97ad09eb") {
+        // Strip "0x97ad09eb" from the input and convert it into a cidv1 - dag-cbor - (sha2-256 : 256)
```

Comment:
why are we stripping this out?  Can we use a named constant for this string and a comment for what it represents?